authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+[id="cco-mode-sts"]
+= Using manual mode with STS
+include::modules/common-attributes.adoc[]
+:context: cco-mode-sts
+toc::[]
+
+Manual mode with STS is available as a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] for Amazon Web Services (AWS).
+
+:FeatureName: Support for AWS Secure Token Service (STS)
+include::modules/technology-preview.adoc[leveloffset=+0]
+
+In manual mode with STS, the individual {product-title} cluster components use AWS Secure Token Service (STS) to assign components IAM roles that provide short-term, limited-privilege security credentials. These credentials are associated with IAM roles that are specific to each component that makes AWS API calls.
+
+Requests for new and refreshed credentials are automated by using an appropriately configured AWS IAM OpenID Connect (OIDC) identity provider, combined with AWS IAM roles. {product-title} signs service account tokens that are trusted by AWS IAM, and can be projected into a pod and used for authentication. Tokens are refreshed after one hour.
+
+.STS authentication flow
+image::142_OpenShift_credentials_STS_0221.svg[Detailed authentication flow between AWS and the cluster when using AWS STS]
+
+Using manual mode with STS changes the content of the AWS credentials that are provided to individual OpenShift components.
+
+.AWS secret format using long-lived credentials
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: <target-namespace> <1>
+  name: <target-secret-name> <2>
+data:
+  aws_access_key_id: <base64-encoded-access-key-id>
+  aws_secret_access_key: <base64-encoded-secret-access-key>
+----
+<1> The namespace for the component.
+<2> The secret name for the component.
+
+.AWS secret format with STS
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: <target-namespace> <1>
+  name: <target-secret-name> <2>
+data:
+  role_name: <operator-role-name> <3>
+  web_identity_token_file: <path-to-token> <4>
+----
+<1> The namespace for the component.
+<2> The secret name for the component.
+<3> The IAM role for the component.
+<4> The path to the service account token inside the pod. By convention, this is `/var/run/secrets/openshift/serviceaccount/token` for {product-title} components.
+
+[id="sts-mode-installing"]
+== Installing an {product-title} cluster configured for manual mode with STS
```

Comment:
Should these be separate modules in the assembly?
```
Using manual mode with STS
* Installing with pre-flight
* Installing manually
* Verifying
```

Reply:
I should think the text below can stay in the assemble allowing you to link to the modules.

Reply:
I was under the impression that I shouldn't create modules unless the content might be used elsewhere - in this case I can't think of another use for it.